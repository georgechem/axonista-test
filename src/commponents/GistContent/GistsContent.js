import React from 'react';
import PropTypes from 'prop-types';

import './gistContent.scss';

import svgDate from './icons/schedule_black_24dp.svg';
import svgUser from './icons/person_black_24dp.svg';
import svgLink from './icons/link_black_24dp.svg';

const GistsContent = ({gist}) => {

    /**
     * Returns properly formatted date OR time - date OR N/A
     * @param dateString
     * @param mode
     * @returns {string}
     */
    const formatDate = (dateString, mode='date') => {
        if(typeof dateString === 'string'){
            const date = new Date(dateString);
            if(date.toString() !== 'Invalid Date'){
                if(mode === 'date'){
                    return `${date.getDate().toString().padStart(2, '0')}/${(date.getMonth() + 1).toString().padStart(2, '0')}/${date.getFullYear()}`;
                }else{
                    return `${date.getHours()}:${date.getMinutes().toString().padStart(2, '0')}:${date.getSeconds().toString().padStart(2, '0')} - 
                ${date.getDate().toString().padStart(2, '0')}/${(date.getMonth() + 1).toString().padStart(2, '0')}/${date.getFullYear()}`;
                }
            }
        }
        return 'N/A';
    }

    return (
        <div className="gistsContent">
            <div className="gistsContent__item">
                <div className="gistsContent__header">
                    <div className="gistsContent__header__bar">
                        <img className="gistsContent__header_svgDate" src={svgDate} alt="time_icon"/>
                        <span>{formatDate(gist?.created_at)}</span>
                    </div>
                    <div className="gistsContent__header__bar">
                        <img src={svgUser} alt="person_icon"/>
                        <span>{gist?.owner?.login}</span>
                    </div>
                </div>
                <hr/>
                <div className="gistContent__body">
                    <div className="gistsContent__body__top">
                        <img className="gistsContent__body__avatar" src={gist?.owner?.avatar_url}/>
                        <div className="gistsContent__body__link">
                            <img src={svgLink}/>
                            <a href={gist?.owner?.html_url} target="_blank">Go to Repository</a>
                        </div>
                    </div>
                    <div className="gistsContent__body__info">
                        <p>updated:{formatDate(gist?.updated_at, 'full')}</p>
                        <div className="gistsContent__body__info__desc">
                            <p><em>{gist?.description}</em></p>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    );
};

GistsContent.propTypes = {
    gist: PropTypes.object.isRequired,
}

export default GistsContent;