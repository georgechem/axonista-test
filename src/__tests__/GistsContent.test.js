import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {cleanup} from '@testing-library/react';
import '@testing-library/jest-dom';
import GistsContent from "../commponents/GistContent/GistsContent";


afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<GistsContent
        gist={{}}/>, div);
});

test('matches Snapshot', () => {

    const tree = renderer.create(<GistsContent
        gist={{}}/>).toJSON();

    expect(tree).toMatchSnapshot();

});