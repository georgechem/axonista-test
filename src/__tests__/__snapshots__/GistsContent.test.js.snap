// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Snapshot 1`] = `
<div
  className="gist"
>
  <table>
    <thead>
      <tr
        className="gist__row__header"
      >
        <th
          className="gist__cell"
        >
          <div
            className="gist__heading"
          >
            <div
              className="gist__heading-text"
            >
              <small>
                Created At
              </small>
            </div>
            <small
              title="created at"
            >
              25/09/2021
            </small>
          </div>
        </th>
        <th
          className="gist__cell"
        >
          <div
            className="gist__heading"
          >
            <div
              className="gist__heading-text"
            >
              <small>
                Avatar
              </small>
            </div>
          </div>
        </th>
        <th
          className="gist__cell"
        >
          <div
            className="gist__heading"
          >
            <div
              className="gist__heading-text"
            >
              <small>
                Files
              </small>
            </div>
          </div>
        </th>
        <th
          className="gist__cell"
        >
          <div
            className="gist__heading"
          >
            <div
              className="gist__heading-text"
            >
              <small>
                Links
              </small>
            </div>
          </div>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="gist__row__body"
      >
        <td>
          <div
            className="gist__body"
          />
        </td>
        <td>
          <div
            className="gist__body"
          >
            <img
              alt="user avatar"
              className="gist__avatar"
              src="https://avatars.githubusercontent.com/u/3243601?v=4"
              title="avatar"
            />
          </div>
        </td>
        <td>
          <div
            className="gist__body"
          >
            <ul
              className="gist__body__list"
            >
              <li
                className="gits__body__list__item"
              >
                <div
                  className="gist__body__list__item-files"
                >
                  <small
                    id="id20"
                    onClick={[Function]}
                  >
                    as
                  </small>
                  <small
                    className="hidden"
                  >
                    language: 
                    n/a
                  </small>
                  <small
                    className="hidden"
                  >
                    size: 
                    0.0kB
                  </small>
                  <small
                    className="hidden"
                  >
                    type: 
                    text/plain
                  </small>
                </div>
              </li>
            </ul>
          </div>
        </td>
        <td>
          <div
            className="gist__body"
          >
            <ul
              className="gist__body__list"
            >
              <li>
                <a
                  href="https://api.github.com/gists/61214c3cd9da6076fc6f443ba310d10a"
                  target="_blank"
                >
                  Gist Url
                </a>
              </li>
              <li>
                <a
                  href="https://api.github.com/gists/61214c3cd9da6076fc6f443ba310d10a/forks"
                  target="_blank"
                >
                  Forks Url
                </a>
              </li>
              <li>
                <a
                  href="https://api.github.com/gists/61214c3cd9da6076fc6f443ba310d10a/commits"
                  target="_blank"
                >
                  Commits URL
                </a>
              </li>
              <li>
                <a
                  href="https://gist.github.com/61214c3cd9da6076fc6f443ba310d10a"
                  target="_blank"
                >
                  Html URL
                </a>
              </li>
              <li>
                <a
                  href="https://api.github.com/users/annakhloyan/repos"
                  target="_blank"
                >
                  Repos URL
                </a>
              </li>
              <li>
                <a
                  href="https://api.github.com/users/annakhloyan"
                  target="_blank"
                >
                  Owner URL
                </a>
              </li>
            </ul>
          </div>
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr
        className="gist__row__footer"
      >
        <td
          className="gist__footer"
        >
          <div
            className="gist__footer-text"
          >
            <small>
              Last update
            </small>
            <span
              title="last updated at"
            >
              <small>
                18:26:24 - 
                25/09/2021
              </small>
            </span>
          </div>
        </td>
        <td>
          <div
            className="gist__footer-text"
          >
            <small>
              Owner login
            </small>
            <span
              title="owner login"
            >
              <small>
                annakhloyan
              </small>
            </span>
          </div>
        </td>
        <td>
          <div />
        </td>
        <td>
          <div />
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
